inspirehep/modules/disambiguation/beard.py

```diff
@@ -0,0 +1,101 @@
+
+    The method receives a list of records and a list of signatures representing
+    records that will be clustered by Beard algorithm.
+
+    The argument 'records' is a list of dictionaries containing values like
+    authors of a particular record, earliest date recorded of the publication
+    and finally publication UUID.
+
+    The argument 'signatures' represents the phonetic block,
+    that is being currently computed by one of 'disambiguation' workers.
+
+    The method dispatches a Celery task to Beard server.
+    Beard algorithm distinguishes different authors for
+    *the same* signature block. The clustered authors are then returned
+    in the format of dictionaries.
+
+    :param records:
+        A list of the records, where at least one author will be
+        clustered (by having the same phonetic block).
+
+        Example:
+            records = [{'authors': [u'Hohm, Olaf', u'Wang, Yi-Nan'],
+                       'publication_id': u'13c3cca8-b0bf-42f5-90d4-...',
+                       'year': u'2015'}]
+
+    :param signatures:
+        A list of signatures belonging to the same signature block,
+        which is currently being clustered.
+
+        Example:
+            signatures =  [{'author_affiliation': u'MIT, Cambridge, CTP',
+                            'author_name': u'Wang, Yi-Nan',
+                            'publication_id': u'13c3cca8-b0bf-42f5-90d4-...',
+                            'signature_id': u'a4156520-4248-a57f-949c361e0dd0',
+                            'author_recid': u'10123',
+                            'author_claimed': False}]
+
+    :return:
+        A tuple containing clusters matched with existing author profiles and
+        clusters for which new profile must be created.
+
+        The first 'bucket' is a list of dictionaries, where each key
+        represents recid of an existing author profile. The second 'bucket'
+        is a list of dictionaries as well, however with enumerated keys,
+        which are meaningless.
+
+        Except clustering, Beard Celery instance is also responsible for
+        matching output of a clustering job with a current state of Inspire
+        system, ie. current links between signatures and their author profiles.
+        The matching process is done using simplex algorithm for maximising
+        overlap between new clusters (Beard output) with signatures clustered
+        by belonging to the same profile.
+
+        To see what is the workflow behind Beard Celery instance, check
+        https://github.com/inspirehep/beard-server
+
+        Example:
+            [{u'10123': [u'a4156520-4248-a57f-949c361e0dd0']}, {}]
+    """
+    # Do not run if any of the two lists are empty.
+    if records and signatures:
+        clusters = celery.current_app.send_task(
+            'beard_server.tasks.make_clusters',
+            (records, signatures),
+            queue=current_app.config.get('DISAMBIGUATION_QUEUE'))
```

Comment:
This how you dispatch a task to another machine with no function definition on the machine that you dispatch the task from.